frontend/hooks/player.ts

```diff
@@ -0,0 +1,33 @@
+import UIfx from "uifx";
+import {
+    useState,
+    useRef,
+    useEffect,
+    Dispatch,
+    SetStateAction,
+    MutableRefObject,
+} from "react";
+
+export function usePlayer(
+    audio: string
+): [
+    boolean,
+    Dispatch<SetStateAction<Boolean>>,
+    MutableRefObject<(() => void) | undefined>
+] {
+    const player = useRef<UIfx>();
+    useEffect(() => {
+        player.current = new UIfx(audio, { throttleMs: 100 });
+    });
+
+    const [notifs, setNotifs] = useState(true);
+
+    const play = useRef<() => void>();
```

Comment:
You can instantiate the function by passing it as a parameter to useRef - this makes sure that play.current is never undefined, and you don't have to do the check everywhere else

Reply:
Remember to also change the types elsewhere so you don't have to do play.current checks anymore